controllers/baremetal_config.go

```diff
@@ -0,0 +1,84 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	metal3iov1alpha1 "github.com/openshift/cluster-baremetal-operator/api/v1alpha1"
+)
+
+const (
+	provisioningNetworkManaged   = "Managed"
+	provisioningNetworkUnmanaged = "Unmanaged"
+	provisioningNetworkDisabled  = "Disabled"
+)
+
+func validateBaremetalProvisioningConfig(config *metal3iov1alpha1.Provisioning) error {
+	if config.Spec.ProvisioningNetwork == "" {
+		config.Spec.ProvisioningNetwork = provisioningNetworkManaged
```

Comment:
Yes, there is a need to have a Config type that wraps the CRD type. Since, these changes have only one instance (support for DHCP external and default behavior) I did not add it in the PR. The containers within metal3 need other config like the ironic inspector endpoint, that would be provided by this wrapper Config type using values in the CRD type. 
Can the Config type be handled in a follow up PR?

Reply:
This is more than a little nit, so I think we should fix it as part of this PR. That may mean a simple function for now, or, since nothing uses the value outside of this function just using a local variable instead of updating the Spec field.